tests/network/bgp/conftest.py

```diff
@@ -0,0 +1,193 @@
+        ipam=libnad.IpamStatic(
+            addresses=[
+                libnad.IpamStatic.Address(address=os.environ["EXTERNAL_FRR_STATIC_IPV4"], gateway=br_ex_gateway_v4)
+            ],
+            routes=[libnad.IpamRoute(dst=DEFAULT_ROUTE_V4.dst, gw=br_ex_gateway_v4)],
+        ),
+    )
+
+    with libnad.NetworkAttachmentDefinition(
+        name="macvlan-nad-bgp",
+        namespace=cnv_tests_utilities_namespace.name,
+        config=libnad.NetConfig(name="macvlan-nad-bgp", plugins=[macvlan_config]),
+        client=admin_client,
+    ) as nad:
+        yield nad
+
+
+@pytest.fixture(scope="module")
+def frr_configmap(
+    workers: list[Node], cnv_tests_utilities_namespace: Namespace, admin_client: DynamicClient
+) -> Generator[ConfigMap]:
+    frr_conf = generate_frr_conf(
+        external_subnet_ipv4=EXTERNAL_SUBNET_IPV4,
+        nodes_ipv4_list=[worker.internal_ip for worker in workers],
+    )
+
+    with ConfigMap(
+        name="frr-config",
+        namespace=cnv_tests_utilities_namespace.name,
+        data={
+            "daemons": (BGP_DATA_PATH / "daemons").read_text(),
+            "frr.conf": frr_conf,
+        },
+        client=admin_client,
+    ) as cm:
+        yield cm
+
+
+@pytest.fixture(scope="module")
+def cluster_network_resource_ra_enabled(
+    network_operator: openshift_nc.Network,
+    admin_client: DynamicClient,
+) -> Generator[None]:
+    with enable_route_advertisements_in_cluster(network_resource=network_operator, client=admin_client):
+        yield
+
+
+@pytest.fixture(scope="module")
+def namespace_cudn(admin_client: DynamicClient) -> Generator[Namespace]:
+    yield from create_udn_namespace(name="test-cudn-bgp-ns", client=admin_client, labels={**CUDN_BGP_LABEL})
+
+
+@pytest.fixture(scope="module")
+def cudn_layer2(namespace_cudn: Namespace) -> Generator[libcudn.ClusterUserDefinedNetwork]:
+    with libcudn.ClusterUserDefinedNetwork(
+        name="l2-network-cudn",
+        namespace_selector=LabelSelector(matchLabels=CUDN_BGP_LABEL),
+        network=libcudn.Network(
+            topology=libcudn.Network.Topology.LAYER2.value,
+            layer2=libcudn.Layer2(
+                role=libcudn.Layer2.Role.PRIMARY.value,
+                ipam=libcudn.Ipam(mode=libcudn.Ipam.Mode.ENABLED.value, lifecycle="Persistent"),
+                subnets=[CUDN_SUBNET_IPV4],
+            ),
+        ),
+        label=APP_CUDN_LABEL,
+    ) as cudn:
+        cudn.wait_for_status_success()
+        yield cudn
+
+
+@pytest.fixture(scope="module")
+def cudn_route_advertisements(
+    cudn_layer2: libcudn.ClusterUserDefinedNetwork,
+    cluster_network_resource_ra_enabled: None,
+    admin_client: DynamicClient,
+) -> Generator[None]:
+    with create_cudn_route_advertisements(
+        name="cudn-route-advertisement", match_labels=APP_CUDN_LABEL, client=admin_client
+    ):
+        yield
+
+
+@pytest.fixture(scope="module")
+def frr_configuration(admin_client: DynamicClient) -> Generator[None]:
```

Comment:
Please choose a different name, `frr_configuration` is also the module name in ocp wrapper which can be confusing and may lead to some problems.